src/sqlfluff/core/parser/segments/base.py

```diff
@@ -115,14 +117,17 @@ def __init__(
         if not pos_marker:
             # If no pos given, it's the pos of the first segment.
             if isinstance(segments, (tuple, list)):
-                pos_marker = PositionMarker.from_child_markers(
-                    *(seg.pos_marker for seg in segments)
-                )
+                if all(hasattr(seg.pos_marker, "source_slice") for seg in segments):
```

Comment:
Is this to allow for `None` values? Could we instead do:
```suggestion
                if all(seg.pos_marker for seg in segments):
```
`hasattr()` is a bit of a confusing way to work around `None` values.